sdk/appconfiguration/spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/stores/ClientStore.java

```diff
@@ -118,43 +119,69 @@ private ConfigurationAsyncClient buildClient(String store) throws IllegalArgumen
     }
 
     /**
-     * Gets the latest Configuration Setting from the revisions given config store that
-     * match the Setting Selector criteria.
+     * Gets the latest Configuration Setting from the revisions given config store that match the Setting Selector
+     * criteria.
      *
-     * @param settingSelector Information on which setting to pull. i.e. number of
-     * results, key value...
+     * @param settingSelector Information on which setting to pull. i.e. number of results, key value...
      * @param storeName Name of the App Configuration store to query against.
      * @return List of Configuration Settings.
      */
     public final ConfigurationSetting getRevison(SettingSelector settingSelector, String storeName) {
+        PagedResponse<ConfigurationSetting> configurationRevision = null;
+        int retryCount = 0;
+        
         ConfigurationAsyncClient client = buildClient(storeName);
```

Comment:
Is there a need to cache these clients?

Reply:
It  should be. It is fixed in 2.0.0. I can move the change here https://github.com/Azure/azure-sdk-for-java/blob/app-config-v2-branch/sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/stores/ClientStore.java.